### ISSUE TITLE
Copy file URL and APIs generate incorrect URLs when configuring file routes 

### ISSUE THREAD

**Issue by MalikBagwala** [payloadcms/payload#1144]
# Bug Report

I have used the official `@payloadcms/plugin-cloud-storage` plugin to upload files directly to an S3 bucket.

This is the plugins array.

```js
plugins: [
    cloudStorage({
      collections: {
        media: {
          prefix: 'cms/media',
          adapter: s3Adapter({
            config: {
              credentials: {
                accessKeyId: process.env.S3_ACCESS_KEY_ID,
                secretAccessKey: process.env.S3_SECRET_ACCESS_KEY
              }
            },
            bucket: process.env.S3_BUCKET
          })
        }
      }
    })
  ]
  ```
  And this is the collection config
  
 ```js
 media: {
    slug: 'media',
    access: {
      read: () => true
    },
    upload: {
      adminThumbnail({ doc }) {
        return `https://yy.com/cms/media/${doc.filename}`
      }
    },
    fields: []
  }
  ```
  
However the `Copy URL` button inside the media copies `http://localhost:3000/media/profile.jpg` And I get the same as response when querying for the data.

![image](https://user-images.githubusercontent.com/23068029/190998069-f836ecca-97d0-4637-a0d7-b34f39ab6c13.png)

 
 
The admin path should work globally or there should be some solution to consistently route all the media via a CDN link

**Comment by jmikrut** [comment]
Hey @MalikBagwala — this is actually expected and is covered in the plugin docs:

https://github.com/payloadcms/plugin-cloud-storage#payload-access-control

By default, the file URLs are intentionally kept the same. This is to preserve Payload access control even while delivering static files. For example, if you have sensitive uploads that should not be read openly by the public, your existing `read` access control on your `upload`-enabled collection is used. 

You can disable this though if you'd like by setting `disablePayloadAccessControl` to `true`. And then at that point, you will see all URLs updated to point directly to your CDN.

Does that make sense?

I am going to close this but am happy to keep the conversation going!

**Comment by MalikBagwala** [comment]
@jmikrut This worked but how do set up a parser middleware.

These are my settings
```
media: {
          disablePayloadAccessControl: true,
          prefix: 'cms/media',
          adapter: s3Adapter({
            config: {
              endpoint: 'https://cdn.xx.com',
              credentials: {
                accessKeyId: process.env.S3_ACCESS_KEY_ID,
                secretAccessKey: process.env.S3_SECRET_ACCESS_KEY
              }
            },
            bucket: "cdn.xx.com"
          })
        }
```

When I upload a file I get the below URL

`https://cdn.xx.com/cdn.xx.com/cms/media/uIsjawkZN5sa.jpg`

Whereas the correct URL should be 

`https://cdn.xx.com/cms/media/uIsjawkZN5sa.jpg`

**Comment by jmikrut** [comment]
Hey @MalikBagwala just posting here as well for reference. We just released `@payloadcms/plugin-cloud-storage@1.0.10` that features a way for you to generate your own custom URLs via passing in the `generateFileURL` property!

**Comment by MalikBagwala** [comment]
> Hey @MalikBagwala just posting here as well for reference. We just released `@payloadcms/plugin-cloud-storage@1.0.10` that features a way for you to generate your own custom URLs via passing in the `generateFileURL` property!

This did the trick @jmikrut
